refactor(csharp/test/Drivers/Interop/Snowflake): Updated the metadata tests to work without the db name

ryan-syed opened the pull request:
Changes:
* Modified tests to check the table type returned in `DriverTests.cs`
* The database and schema name is not required for the getting the metadata anymore. The metadata calls in the Driver prefix the specific DB name whose INFORMATION_SCHEMA tables need to be accessed.

ryan-syed commented:
Similar to PR: https://github.com/apache/arrow-adbc/pull/1351, except for `populateMetadata`, `prepareDbSchemasSQL`, `prepareTablesSQL`, and `prepareColumnsSQL`.

The implementation is based on proposal: https://github.com/apache/arrow-adbc/issues/1332##Proposal and improving existing API implementation.

jduo commented on go/adbc/driver/snowflake/connection.go:
```diff
@@ -238,84 +239,59 @@ func (c *cnxn) GetInfo(ctx context.Context, infoCodes []adbc.InfoCode) (array.Re
-	cond := strings.Join(conditions, " AND ")
-
 	result = make(map[string][]string)
+	uniqueCatalogSchema := make(map[string]map[string]bool)
```
It is hard to tell what these maps are being used for. Some comments would be helpful.

ryan-syed replied:
Those changes have already been merged. The map is used find unique schemas for a catalog, so that duplicate schema entries are not created.

ryan-syed commented:
Functionality changes seems to be merged already in PR: https://github.com/apache/arrow-adbc/pull/1338. Some trivial changes for `DriverTests.cs` are pending.

davidhcoe commented on csharp/test/Drivers/Interop/Snowflake/DriverTests.cs:
```diff
@@ -83,12 +83,6 @@ public DriverTests()
             Dictionary<string, string> options = new Dictionary<string, string>();
             _snowflakeDriver = SnowflakeTestingUtils.GetSnowflakeAdbcDriver(_testConfiguration, out parameters);
 
-            string databaseName = _testConfiguration.Metadata.Catalog;
```
everything passes with these values removed?

ryan-syed replied:
The database name is not required for the getting the metadata anymore. The calls prefix the specific DB name whose INFORMATION_SCHEMA tables need to be accessed.

CurtHagenlocher commented:
So the only material changes remaining are in the test?

lidavidm commented:
If so do you mind reverting the spurious Go changes and updating the PR title and description?

ryan-syed commented:
> If so do you mind reverting the spurious Go changes and updating the PR title and description?

I have removed the whitespace changes and updated the title and description.
